server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -740,6 +742,122 @@ public boolean accept(File file) {
+                        context.put("resourceFile", resourceFile.getAbsolutePath());
+
+                        String command = templateExecutor.execute(platformConfig.aapt2compileCmd, context);
+                        processExecutor.execute(command);
+                    }
+                }
+            }
+
+        } catch (IOException | InterruptedException e) {
+            throw new ExtenderException(e, processExecutor.getOutput());
+        }
+
+        return outputDirectory;
+    }
+
+    private Map<String, File> linkAndroidResources(File compiledResourcesDir, String platform, Map<String, Object> mergedAppContext) throws ExtenderException {
+        LOGGER.info("Linking Android resources");
+
+        Map<String, Object> context = createContext();
+        Map<String, File> files = new HashMap<>();
+
+        try {
+            // write compiled resource list to a txt file
+            StringBuilder sb = new StringBuilder();
+            for (File packageDir : compiledResourcesDir.listFiles(File::isDirectory)) {
+                for (File file : packageDir.listFiles()) {
+                    if (file.getAbsolutePath().endsWith(".flat")) {
+                        sb.append(file.getAbsolutePath() + " ");
+                    }
+                }
+            }
+            File resourceList = new File(buildDirectory, "compiledresources.txt");
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(resourceList))) {
+                writer.write(sb.toString());
+            }
+            context.put("resourceListFile", resourceList.getAbsolutePath());
+
+            // extra packages
+            if (mergedAppContext.containsKey("aaptExtraPackages")) {
+                context.put("extraPackages", String.join(":", (List<String>)mergedAppContext.get("aaptExtraPackages")));
+            }
+
+            File manifestFile = new File(buildDirectory, MANIFEST_ANDROID);
+            context.put("manifestFile", manifestFile.getAbsolutePath());
+
+            File resourceIdsFile = new File(buildDirectory, "resource_ids.txt");
+            context.put("resourceIdsFile", resourceIdsFile.getAbsolutePath());
+
+            File outputJavaDirectory = createDir(buildDirectory, "out_java");
+            context.put("outJavaDirectory", outputJavaDirectory.getAbsolutePath());
+
+            File outApkFile = new File(buildDirectory, "compiledresources.apk");
+            context.put("outApkFile", outApkFile.getAbsolutePath());
+
+            files.put("resourceIdsFile", resourceIdsFile);
+            files.put("outApkFile", outApkFile);
+            files.put("outJavaDirectory", outputJavaDirectory);
+
+            String command = templateExecutor.execute(platformConfig.aapt2linkCmd, context);
+            processExecutor.execute(command);
+        }
+        catch (IOException | InterruptedException e) {
+            throw new ExtenderException(e, processExecutor.getOutput());
+        }
+
+        return files;
+    }
+
```

Comment:
We should move the android functions (later) to an android specific module, since the Extender.java is really starting to be quite large.